src/Microsoft.DotNet.Tools.Pack/InputOutputPathsCalculator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Tools.Pack
+{
+    public class InputOutputPathsCalculator
+    {
+        private readonly Project _project;
+
+        private readonly string _configuration;
+
+        private bool OutputPathSet => !string.IsNullOrWhiteSpace(OutputPathParameter);
+
+        public string InputPathParameter { get; }
+
+        public string OutputPathParameter { get; }
+
+        public bool InputPathSet => !string.IsNullOrWhiteSpace(InputPathParameter);        
+
+        public string InputPath => InputPathSet ? InputPathParameter : OutputPath;
+
+        public string OutputPath
+        {
+            get
+            {
+                if (OutputPathSet)
+                {
+                    return OutputPathParameter;
+                }
+
+                var outputPath = Path.Combine(
+                    _project.ProjectDirectory,
+                    Constants.BinDirectoryName,
+                    _configuration);
+
+                return outputPath;
+            }
+        }        
+
+        public InputOutputPathsCalculator(Project project, string inputPath, string outputPath, string configuration)
+        {
+            _project = project;
+            InputPathParameter = inputPath;
+            OutputPathParameter = outputPath;
+            _configuration = configuration;
+        }
+
+        public string InputPathForContext(ProjectContext context)
+        {
+            return OutputPathSet ? OutputPath : Path.Combine(InputPath, context.TargetFramework.GetTwoDigitShortFolderName());
```

Comment:
why is InputPathForContext returning an OutputPath? These names are very confusing.